src/SimpleGraphs/generators/staticgraphs.jl

```diff
@@ -536,6 +536,54 @@ function double_binary_tree(k::Integer)
     return g
 end
 
+"""
+    regular_tree(k::Integer, z::integer)
+
+Create a [regular tree](https://en.wikipedia.org/wiki/Bethe_lattice), 
+also known as Bethe Lattice or Cayley Tree,
+of depth `k` and degree `z`.
+
+# Examples
+```jldoctest
+julia> regular_tree(4, 3)
+{40, 39} undirected simple Int64 graph
+
+julia> regular_tree(5, 2) == binary_tree(5)
+true
+```
+"""
+function regular_tree(k::T, z::T) where {T<:Integer}
+    k <= 0 && return SimpleGraph(0)
+    k == 1 && return SimpleGraph(1)
```

Comment:
In these cases we return a graph of of eltype `Int`, but in all other cases we return a graph of type `T`.

Reply:
To be completely honest, I am not sure if it is the best idea to determine the eltype of the graphs from `k` and `z` - we do that indeed for other graphs here, but I never really liked that.

Reply:
Maybe we can have a function signature
```julia
regular_tree(T::Type{<:Integer}, k::Integer, z::Integer)
```
and then define something like
```julia
regular_tree(k::Integer, z::Integer) = regular_tree(Int64, k, z)
```
that for be similar to the function `sprand` from `SparseArrays` for example.